src/Microsoft.Health.Dicom.Api/Features/Filters/QueryModelStateValidatorAttribute.cs

```diff
@@ -17,7 +18,8 @@ public override void OnActionExecuting(ActionExecutingContext context)
             EnsureArg.IsNotNull(context, nameof(context));
             if (!context.ModelState.IsValid)
             {
-                throw new InvalidQueryStringValuesException(context.ModelState.FirstOrDefault(x => x.Value.Errors.Any()).Key);
+                (string key, ModelStateEntry value) = context.ModelState.Where(x => x.Value.Errors.Count > 0).First();
+                throw new InvalidQueryStringValuesException(key, value.Errors[0].ErrorMessage);
```

Comment:
I like this change, it make result more clear to customer.

Reply:
I see you did something similar haha